Backend/src/main/java/lt/vu/ads/controllers/OrderController.java

```diff
@@ -0,0 +1,37 @@
+package lt.vu.ads.controllers;
+
+import lt.vu.ads.exceptions.ResourceNotFoundException;
+import lt.vu.ads.models.Order;
+import lt.vu.ads.repositories.OrderRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import java.util.List;
+
+@RestController
+@RequestMapping("/api")
+public class OrderController {
+    @Autowired
+    OrderRepository orderRepository;
+
+    @GetMapping("/orders")
+    public List<Order> getAllOrders(){
+        return orderRepository.findAll();
+    }
+
+    @GetMapping("/order/{id}")
+    public ResponseEntity < Order > getOrderById(@PathVariable(value = "id") Long orderId)
+            throws ResourceNotFoundException {
+        Order order = orderRepository.findById(orderId)
+                .orElseThrow(() -> new ResourceNotFoundException("Order not found for this id :: " + orderId));
+        return ResponseEntity.ok().body(order);
+    }
+
+    @PostMapping("/orders")
+    public Order createOrder(@RequestBody Order order)
+    {
+        return orderRepository.save(order);
```

Comment:
There are no checkups for order class. Since we cannot fully trust what user writes in fields, we need to validate them. For example: convenient arrival time should be checked: convenientArrivalTimeFrom must be earlier than convenientArrivalTimeTo.

Also user can't assign adress_id to the source (or desination) address. While creating order user in the UI inputs all fields related to Address class (city, street,...) but in the API code should check if that address exists in Address table or not. If not, new record inserted. If record exists, it's id should be assigned to that order.

Also orderCode generates our system, not user, so maybe here should be called some other class which generates it (for now, class can have empty methods, later on we will implement them).
Same with courier.

Reply:
Does the user specify convenientArrivalTimeFrom and convenientArrivalTimeTo when registering an order? I think these fields should only be checked if the user wants to change the arrival time. I mean only in put method.

Reply:
Ah yeah, my bad ^^ convenient arrival time should be in put method